airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/destination/record_buffer/BaseSerializedBuffer.kt

```diff
@@ -42,7 +42,7 @@ protected constructor(private val bufferStorage: BufferStorage) : SerializableBu
      */
     @Deprecated("")
     @Throws(IOException::class)
-    protected abstract fun writeRecord(record: AirbyteRecordMessage)
+    protected abstract fun writeRecord(record: AirbyteRecordMessage, generationId: Long = 0)
```

Comment:
I considered that, as well as changes necessary to make my code use the new methods.

I think that's the correct design. Threading things through is an anti pattern.

But the even more correct pattern is for this code to know nothing about airbyte fields, which I imagine is how it will work in a new CDK. So I punted on doing the work to rewrite anything.

Reply:
Actually, is it constant per sync or per stream?

Reply:
ah yeah, it's per stream. Isn't this object instanced per stream? (totally fair that this would be too much work for no payoff though)